RadarSDK/RadarVerificationManager.m

```diff
@@ -182,57 +182,47 @@ - (void)trackVerifiedWithBeacons:(BOOL)beacons completionHandler:(RadarTrackVeri
+        NSTimeInterval minInterval = self.startedInterval;
         
-        [self trackVerifiedWithBeacons:beacons completionHandler:^(RadarStatus status, RadarVerifiedLocationToken *_Nullable token) {
-            NSTimeInterval expiresIn = 0;
-            NSTimeInterval minInterval = interval;
+        if (token) {
+            expiresIn = token.expiresIn;
             
-            if (token) {
-                expiresIn = token.expiresIn;
-                
-                // if expiresIn is shorter than interval, override interval
-                minInterval = MIN(expiresIn, interval);
-            }
-            
-            // re-request early to maximize the likelihood that a cached token is available
-            if (minInterval > 20) {
-                minInterval = minInterval - 10;
-            }
-            
-            // min interval is 10 seconds
-            if (minInterval < 10) {
-                minInterval = 10;
-            }
-            
-            if (self.scheduled) {
-                [[RadarLogger sharedInstance] logWithLevel:RadarLogLevelDebug message:@"Token request already scheduled"];
-                
-                return;
-            }
-            
-            [[RadarLogger sharedInstance] logWithLevel:RadarLogLevelDebug message:[NSString stringWithFormat:@"Requesting token again in %f seconds | minInterval = %f; expiresIn = %f; interval = %f", minInterval, minInterval, expiresIn, interval]];
-            
-            self.scheduled = YES;
-            
-            dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(minInterval * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
-                if (self.started) {
-                    [[RadarLogger sharedInstance] logWithLevel:RadarLogLevelDebug message:@"Token request interval fired"];
-                    
-                    weakTrackVerified();
-                    
-                    self.scheduled = NO;
-                }
-            });
-        }];
-    };
+            // if expiresIn is shorter than interval, override interval
+            minInterval = MIN(expiresIn, self.startedInterval);
+        }
+        
+        // re-request early to maximize the likelihood that a cached token is available
+        if (minInterval > 20) {
+            minInterval = minInterval - 10;
+        }
+        
+        // min interval is 10 seconds
+        if (minInterval < 10) {
+            minInterval = 10;
+        }
+        
+        [NSObject cancelPreviousPerformRequestsWithTarget:self selector:@selector(intervalFired) object:nil];
+        
+        [[RadarLogger sharedInstance] logWithLevel:RadarLogLevelDebug message:[NSString stringWithFormat:@"Requesting token again in %f seconds | minInterval = %f; expiresIn = %f; startedInterval = %f", minInterval, minInterval, expiresIn, self.startedInterval]];
+        
+        [self performSelector:@selector(intervalFired) withObject:nil afterDelay:minInterval];
```

Comment:
I think there might be an edge case here where if `stopTrackingVerified` is called right after the `intervalFired` is starting, the `intervalFired` function will still run, and call this `callTrackVerified` and continue the loop. 
(depending on how iOS does time slicing and threading of the calls)

Reply:
I can reproduce this with a low tracking interval with the example app. if I click(call) stopTrackingVerified at certain times, it doesn't stop the tracking loop

Reply:
Good catch, I added back a `started` flag to cover this